gpameter.user.js

```diff
@@ -27,9 +27,15 @@ var grade_values = {
 };
 console.log(studentId);
 add_checkboxes = function(){
+	var mapElems = new Map();
 	$(".cgpa_cal_check").remove();
-	elems = $(".hierarchyLi.dataLi").not(".hierarchyHdr, .hierarchySubHdr");	
+	elems = $(".hierarchyLi.dataLi").not(".hierarchyHdr, .hierarchySubHdr");
 	elems.each(function(i){
+		var cname = $(this).children(".col2").html().trim();
```

Comment:
Is this course name or course id?
Let's prefer using course id since course names may change across semesters while course ids would remain the same.

Also, let's rename the variable to course_id instead.